### PR TITLE
Systemd stage 1: Unhide docs and round out assertions about unsupported features.

### PR DESCRIPTION
**Pull request by ElvishJerricco**
I think 23.05 would be a good time to unhide the systemd initrd options.

### CODE REVIEW

**Comment by ElvishJerricco** [comment]
My only concern is that it does add a lot of sort of redundant things to the options list, since it repeats all the systemd options yet again. If anyone knows how to show the top level things like `boot.initrd.systemd.services` but hide the child options, that would probably be good.

File: nixos/modules/services/networking/iscsi/root-initiator.nix [hunk 1]
```diff
@@ -185,6 +185,10 @@ in
         assertion = cfg.loginAll -> cfg.target == null;
         message = "iSCSI target name is set while login on all portals is enabled.";
       }
+      {
+        assertion = !config.boot.initrd.systemd.enable;
+        message = "systemd stage does not support iscsi yet.";
```
Comment (lilyinstarlight) [inline nixos/modules/services/networking/iscsi/root-initiator.nix:190]:
```suggestion
        message = "systemd stage 1 does not support iscsi yet.";
```

I think this was slightly typo'd

Comment (ElvishJerricco) [reply nixos/modules/services/networking/iscsi/root-initiator.nix]: Thanks.

**Comment by ElvishJerricco** [comment]
Ok, I've set the systemd unit options to `visible = "shallow";`, and now they show up without their child options. That's perfect.

**Review by arianvp — APPROVED** [review]

**Comment by ElvishJerricco** [comment]
@arianvp I wasn't expecting the networkd PR to be merged first! :P There's a couple of things to change here now.

1. The network module should no longer claim to be incompatible.
2. I should also make the `boot.initrd.systemd.network` options shallow to some extent.

**Comment by flokli** [comment]
@ElvishJerricco can you rebase this on top of current master, and undraft when it's ready for review again?

**Comment by ElvishJerricco** [comment]
I'm not *exactly* happy with the ugly changes to `networkd.nix`, so please let me know if anyone has a better idea for how to do that.

**Comment by ElvishJerricco** [comment]
Ok I think that's a little better for networkd.

/cc @arianvp

**Comment by maxeaubrey** [comment]
I've been happily running systemd in stage 1 for a few weeks now, but with this PR I can't rebuild my system with it enabled? Is that intentional, and I shouldn't have been running systemd stage 1 yet when using zfs?

```
error:
       Failed assertions:
       - systemd stage 1 does not support 'boot.initrd.postDeviceCommands'. Please
       convert it to analogous systemd units in 'boot.initrd.systemd'.

         Definitions:
         - /home/systems/nixpkgs/nixos/modules/tasks/filesystems/zfs.nix

       - systemd stage 1 does not support 'boot.initrd.extraUtilsCommands'. Please
       convert it to analogous systemd units in 'boot.initrd.systemd'.

         Definitions:
         - /home/systems/nixpkgs/nixos/modules/tasks/filesystems/zfs.nix

       - systemd stage 1 does not support 'boot.initrd.extraUtilsCommandsTest'. Please
       convert it to analogous systemd units in 'boot.initrd.systemd'.

         Definitions:
         - /home/systems/nixpkgs/nixos/modules/tasks/filesystems/zfs.nix
(use '--show-trace' to show detailed location information)
```

**Comment by flokli** [comment]
This PR added some assertions, because systemd in initrd actually ignores some of these keys.

What are these keys set to in your case?

**Comment by lilyinstarlight** [comment]
@flokli, those are being set automatically by the ZFS module so I think that module needs to be updated to conditionalize some of the `boot.initrd.` settings it does based on `boot.initrd.systemd.enable`

**Comment by maxeaubrey** [comment]
Sorry, should have added there context. Exactly what @lilyinstarlight said, this is just a side effect of having the zfs module enabled, nothing special on my side.

**Comment by ElvishJerricco** [comment]
Ok I thought I had checked the `boot.initrd.*Commands` things better than I did. We should probably revert that specific change and go over it with a more fine tooth comb, preferably but not necessarily before 23.05.

**Pull request merged**